modular_skyrat/modules/alt_vox/code/vox_procs.dm

```diff
@@ -0,0 +1,178 @@
+	<LI>Sound effects begin with an 's' before the actual word, e.g. scensor</LI>
+	<LI>Use Ctrl+F to see if a word exists in the list.</LI></UL><HR>
+	"}
+	switch(vox_type)
+		if(VOX_NORMAL)
+			var/index = 0
+			for(var/word in GLOB.vox_sounds)
+				index++
+				dat += "<A href='?src=[REF(src)];say_word=[word]'>[capitalize(word)]</A>"
+				if(index != GLOB.vox_sounds.len)
+					dat += " / "
+		if(VOX_HL)
+			var/index = 0
+			for(var/word in GLOB.vox_sounds_hl)
+				index++
+				dat += "<A href='?src=[REF(src)];say_word=[word]'>[capitalize(word)]</A>"
+				if(index != GLOB.vox_sounds_hl.len)
+					dat += " / "
+
+	var/datum/browser/popup = new(src, "announce_help", "Announcement Help", 500, 400)
+	popup.set_content(dat)
+	popup.open()
+
+
+/mob/living/silicon/ai/proc/announcement()
+	var/static/announcing_vox = 0 // Stores the time of the last announcement
+	if(announcing_vox > world.time)
+		to_chat(src, "<span class='notice'>Please wait [DisplayTimeText(announcing_vox - world.time)].</span>")
+		return
+
+	var/message = input(src, "WARNING: Misuse of this verb can result in you being job banned. More help is available in 'Announcement Help'", "Announcement", src.last_announcement) as text|null
+
+	if(!message || announcing_vox > world.time)
+		return
+
+	last_announcement = message
+
+	if(incapacitated())
+		return
+
+	if(control_disabled)
+		to_chat(src, "<span class='warning'>Wireless interface disabled, unable to interact with announcement PA.</span>")
+		return
+
+	var/list/words = splittext(trim(message), " ")
+	var/list/incorrect_words = list()
+
+	if(words.len > 30)
+		words.len = 30
+
+	switch(vox_type)
+		if(VOX_NORMAL)
+			for(var/word in words)
+				word = lowertext(trim(word))
+				if(!word)
+					words -= word
+					continue
+				if(!GLOB.vox_sounds[word])
+					incorrect_words += word
+		if(VOX_HL)
+			for(var/word in words)
+				word = lowertext(trim(word))
+				if(!word)
+					words -= word
+					continue
+				if(!GLOB.vox_sounds_hl[word])
+					incorrect_words += word
+
+	if(incorrect_words.len)
+		to_chat(src, "<span class='notice'>These words are not available on the announcement system: [english_list(incorrect_words)].</span>")
+		return
+
+	announcing_vox = world.time + VOX_DELAY
+
+	log_game("[key_name(src)] made a vocal announcement with the following message: [message].")
+	log_talk(message, LOG_SAY, tag="VOX Announcement")
+
+	for(var/word in words)
+		play_vox_word(word, src.z, null, vox_type)
+
+
+/proc/play_vox_word(word, z_level, mob/only_listener, vox_type)
+
+	word = lowertext(word)
+	switch(vox_type)
+		if(VOX_NORMAL)
+			if(GLOB.vox_sounds[word])
+
+				var/sound_file = GLOB.vox_sounds[word]
+				var/sound/voice = sound(sound_file, wait = 1, channel = CHANNEL_VOX)
+				voice.status = SOUND_STREAM
+
+			// If there is no single listener, broadcast to everyone in the same z level
+				if(!only_listener)
+					// Play voice for all mobs in the z level
+					for(var/mob/M in GLOB.player_list)
```

Comment:
single letter bad

Reply:
That's how TG had it, It's just a copy paste

Reply:
Copy paste bad 😔